### ISSUE TITLE
Option to disable case-sensitive library sorting?

### ISSUE THREAD

**Issue by therealfumbles** [fooyin/fooyin#233]
It would be nice if there was an option to have the library tree not sort itself as case-sensitive. 

Currently, any artist whose name starts with a lower-case or accented letter is sorted to the end of the library tree.

**Comment by ludouzi** [comment]
The case-sensitivity of all sorting defaults to the system locale, which in most cases is case-insensitive. The exception is the C locale:

> Note: In the C locale, when case-sensitive, all lower-case letters sort after all upper-case letters, where most locales sort each lower-case letter either immediately before or immediately after its upper-case partner. Thus "Zap" sorts before "ape" in the C locale but after in most others. https://doc.qt.io/qt-6/qcollator.html#caseSensitivity

I'll investigate whether setting a different default locale is a good idea.

**Comment by ludouzi** [comment]
Actually, I'm guessing your language is set to system default. Does the library tree sort correctly if you set it to something else?

**Comment by therealfumbles** [comment]
Yep, setting it to US or British English sorts it correctly. Wouldn't have suspected the language to be the reason!

**Comment by ludouzi** [comment]
Yeah, it's because the C locale defaults to ASCII ordering (A-Z then a-z). This may also be a problem for systems where Qt was compiled without ICU support.

I think the path forward is to implement a custom sorting algorithm using ICU directly since we depend on it.
